### PR TITLE
Document Microsoft.Testing.Platform

### LINKED ISSUE #2298: Change the name of the MSTest.Runner
## Summary

Change the name of the MSTest.Runner to Microsoft.Testing.Runner  (or something else that is test framework agnostic.

## Background and Motivation

The new MSTest.Runner is intended to also support the other frameworks, XUnit and NUnit, ref https://github.com/microsoft/testfx/issues/2164

It is then very confusing and unfortunate to have this being named MSTest.Runner, as it is more of a Microsoft.Testing.Runner.

It also makes comments like this

> We just need to provide enough new nice features for you to want to migrate to MSTest

in https://github.com/microsoft/testfx/issues/2164#issuecomment-1910743166

confusing.  Does the [commenter](https://github.com/Evangelink)  mean that this is an attempt to convert all XUnit and NUnit users to move to use the MSTest framework or does he mean that he want users to use the misnamed MSTest.Runner instead of the current runners ?   I prefer to believe it is the latter.  If it is the former, well, the runner is not that important, the framework is.

## Proposed Feature

Change the name of the MSTest.Runner to Microsoft.Testing.Runner  (or something else that is test framework agnostic.

### PR DESCRIPTION
**Pull request by Evangelink**
## Summary

Extract from MSTest runner docs elements that are common to all test frameworks built on Microsoft.Testing.Platform

Fixes https://github.com/microsoft/testfx/issues/2298

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/testing/unit-testing-mstest-runner-intro.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/core/testing/unit-testing-mstest-runner-intro.md) | [MSTest runner overview](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-mstest-runner-intro?branch=pr-en-us-40043) |
| [docs/core/testing/unit-testing-mstest-sdk.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/core/testing/unit-testing-mstest-sdk.md) | [MSTest SDK overview](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-mstest-sdk?branch=pr-en-us-40043) |
| [docs/core/testing/unit-testing-platform-exit-codes.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/core/testing/unit-testing-platform-exit-codes.md) | [Microsoft.Testing.Platform exit codes](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-exit-codes?branch=pr-en-us-40043) |
| [docs/core/testing/unit-testing-platform-extensions.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/core/testing/unit-testing-platform-extensions.md) | [docs/core/testing/unit-testing-platform-extensions](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions?branch=pr-en-us-40043) |
| [docs/core/testing/unit-testing-platform-integration-dotnet-test.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/core/testing/unit-testing-platform-integration-dotnet-test.md) | [Use Microsoft.Testing.Platform with `dotnet test`](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-integration-dotnet-test?branch=pr-en-us-40043) |
| [docs/core/testing/unit-testing-platform-intro.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/core/testing/unit-testing-platform-intro.md) | [Microsoft.Testing.Platform overview](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-intro?branch=pr-en-us-40043) |
| [docs/core/testing/unit-testing-platform-runsettings.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/core/testing/unit-testing-platform-runsettings.md) | [Use runsettings with Microsoft.Testing.Platform](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-runsettings?branch=pr-en-us-40043) |
| [docs/core/testing/unit-testing-platform-telemetry.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/core/testing/unit-testing-platform-telemetry.md) | [docs/core/testing/unit-testing-platform-telemetry](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-telemetry?branch=pr-en-us-40043) |
| [docs/core/testing/unit-testing-platform-vs-vstest.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/core/testing/unit-testing-platform-vs-vstest.md) | [Microsoft.Testing.Platform and VSTest comparison](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-vs-vstest?branch=pr-en-us-40043) |
| [docs/whats-new/dotnet-docs-mod0.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/whats-new/dotnet-docs-mod0.md) | [".NET docs: What's new for December 2023"](https://review.learn.microsoft.com/en-us/dotnet/whats-new/dotnet-docs-mod0?branch=pr-en-us-40043) |
| [docs/whats-new/dotnet-docs-mod1.md](https://github.com/dotnet/docs/blob/66d614af05f772535aa23fb21a71a4812b80e353/docs/whats-new/dotnet-docs-mod1.md) | [.NET docs: What's new for January 2024](https://review.learn.microsoft.com/en-us/dotnet/whats-new/dotnet-docs-mod1?branch=pr-en-us-40043) |

</details>

### CODE REVIEW

**Comment by Evangelink** [comment]
I see it's complaining about the links rename and asking to add some redirection but I am not sure how this is done here. Could you help me?

**Comment by BillWagner** [comment]
Hi @Evangelink 

I haven't reviewed all the changes closely yet, but you need to configure redirections for any of the files you renamed or removed. That way, once this goes live, any inbound links will redirect to the updated content.

**Comment by Evangelink** [comment]
@BillWagner Could you let me know how to do so?

**Comment by BillWagner** [comment]
@Evangelink 

You need to add configs to the [redirection file](https://github.com/dotnet/docs/blob/main/.openpublishing.redirection.core.json) for the moved files. By convention, we sort these entries by source path to avoid merge conflicts.

File: docs/core/testing/unit-testing-platform-intro.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+title: Microsoft.Testing.Platform overview
+description: Learn about Microsoft.Testing.Platform, a lightweight way to run tests without depending on the .NET SDK.
+author: Evangelink
+ms.author: amauryleve
+ms.date: 03/17/2024
+---
+
+# Microsoft.Testing.Platform overview
+
+Microsoft.Testing.Platform is a lightweight and portable alternative to [VSTest](https://github.com/microsoft/vstest) for running tests in all contexts (continuous integration (CI) pipelines, CLI, Visual Studio Test Explorer, VS Code Text Explorer...). The Microsoft.Testing.Platform is embedded directly in your test projects, and there's no other app dependencies, such as `vstest.console` or `dotnet test` needed to run your tests.
```
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-intro.md:11]:
```suggestion
Microsoft.Testing.Platform is a lightweight and portable alternative to [VSTest](https://github.com/microsoft/vstest) for running tests in all contexts, including continuous integration (CI) pipelines, CLI, Visual Studio Test Explorer, and VS Code Text Explorer. The Microsoft.Testing.Platform is embedded directly in your test projects, and there's no other app dependencies, such as `vstest.console` or `dotnet test` needed to run your tests.
```

File: docs/core/testing/unit-testing-platform-vs-vstest.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+### Microsoft.Testing.Platform namespaces
+
+Microsoft.Testing.Platform is based on [Microsoft.Testing.Platform](https://www.nuget.org/packages/Microsoft.Testing.Platform) NuGet package and other libraries in the `Microsoft.Testing.*` namespace. Like VSTest, the `Microsoft.Testing.Platform` is open-source and has a [microsoft/testfx](https://github.com/microsoft/testfx/tree/main/src/Platform/Microsoft.Testing.Platform) GitHub repository.
+
+## Communication protocol
+
+> [!NOTE]
+> Test Explorer supports the Microsoft.Testing.Platform protocol from version 17.10. If you run/debug your tests using earlier versions of Visual Studio,
+> Test Explorer will use `vstest.console.exe` and the old protocol to run these tests.
```
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-vs-vstest.md:43]:
```suggestion
> The Visual Studio Test Explorer supports the Microsoft.Testing.Platform protocol from version 17.10 onward. If you run/debug your tests using earlier versions of Visual Studio, Test Explorer will use `vstest.console.exe` and the old protocol to run these tests.
```

**Review by IEvangelist — APPROVED** [review]

**Comment by IEvangelist** [comment]
This looks great, but we'll need to get these build errors/warnings cleaned up. [Follow this link, and look at each line number and its error](https://buildapi.docs.microsoft.com/Output/PullRequest/a3bda507-3390-de91-8f7f-26f90f4e5fc8/202403181000016564-40043/BuildReport?accessString=cb3057e21d5fd8d99857eca4df6036626004b5ea2e8abc3428dcd001c412e031).

**Comment by Evangelink** [comment]
> @Evangelink
> 
> You need to add configs to the [redirection file](https://github.com/dotnet/docs/blob/main/.openpublishing.redirection.core.json?rgh-link-date=2024-03-18T15%3A26%3A11Z) for the moved files. By convention, we sort these entries by source path to avoid merge conflicts.

Thanks @BillWagner! I have fixed the redirections. @IEvangelist Done :)

**Pull request merged**